resilience4j-spring/src/main/java/io/github/resilience4j/timelimiter/configure/TimeLimiterAspect.java

```diff
@@ -139,8 +143,17 @@ private static TimeLimiter getTimeLimiterAnnotation(ProceedingJoinPoint proceedi
         }
     }
 
-    private static Object handleJoinPointCompletableFuture(
-            ProceedingJoinPoint proceedingJoinPoint, io.github.resilience4j.timelimiter.TimeLimiter timeLimiter) throws Throwable {
+    private static Object handleJoinPointFuture(
+            ProceedingJoinPoint proceedingJoinPoint, io.github.resilience4j.timelimiter.TimeLimiter timeLimiter, boolean blocking) throws Throwable {
+        if(blocking) {
+            return timeLimiter.executeFutureSupplier(() -> {
```

Comment:
I think this doesn't work.
The method which is intercepted returns a `Future`, but the Aspect returns a value. 
I think the Aspect should/must return the same type like the intercepted method.

Reply:
Oh yes. My bad. I just updated the PR accordingly, so that it is similar to https://resilience4j.readme.io/docs/timeout#decorate-and-execute-a-functional-interface